brave/src/main/java/brave/Tracer.java

```diff
@@ -261,6 +263,17 @@ public SpanInScope withSpanInScope(Span span) {
     return currentContext != null ? toSpan(currentContext) : null;
   }
 
+  /**
+   * Conditionally joins a span, or starts a new trace, depending on if a trace context was
+   * extracted from the carrier (usually an incoming request).
+   */
+  @Nullable public <C> Span nextSpan(Extractor<C> extractor, C carrier) {
```

Comment:
using nextSpan though not sure if that's an awesome name. can't think of a better one.